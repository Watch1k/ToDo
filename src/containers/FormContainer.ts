import { connect } from 'react-redux'
import { addTodo } from '../actions'
import ToDoForm from '../components/Form/Form'
import { bindActionCreators, Dispatch } from 'redux'
import { IState } from '../interfaces/state'

const mapStateToProps = (state: IState) => ({
  todos: state.todos,
})

const mapDispatchToProps = (dispatch: Dispatch) => ({
  ...bindActionCreators({ addTodo }, dispatch),
})

export type IToDoFormProps = ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>

export default connect(mapStateToProps, mapDispatchToProps)(ToDoForm)
