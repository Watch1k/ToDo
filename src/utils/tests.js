import React from 'react'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react'

export const renderWithRedux = (ui, store) => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store,
})
