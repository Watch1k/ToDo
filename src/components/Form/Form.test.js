import React from 'react'
import { fireEvent } from '@testing-library/react'
import { store } from '../../store'
import { renderWithRedux } from '../../utils/tests'
import FormContainer from '../../containers/FormContainer'

describe('Form component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = renderWithRedux(<FormContainer />, store)
  })

  it('Renders', () => {
    expect(wrapper.getByTestId('form')).toBeInTheDocument()
  })

  it('Matches snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('Input works', () => {
    const input = wrapper.getByTestId('form-input')
    const value = 'New task'

    fireEvent.change(input, { target: { value } })
    expect(input.value).toBe(value)
  })

  it('Submit adds new task and prevent if empty', () => {
    const input = wrapper.getByTestId('form-input')
    const value = 'New task'

    fireEvent.change(input, { target: { value } })
    fireEvent.submit(wrapper.getByTestId('form'))

    expect(store.getState().todos.some(_ => _.text === value)).toBe(true)

    // Prevent if empty
    const getLength = () => store.getState().todos.length
    const length = getLength()
    fireEvent.change(input, { target: { value: '' } })
    fireEvent.submit(wrapper.getByTestId('form'))

    expect(getLength()).toBe(length)
  })
})
