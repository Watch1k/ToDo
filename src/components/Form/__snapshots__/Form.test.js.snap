// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component Matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-inline"
    data-testid="form"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control-wrapper"
      >
        <div
          class="ant-form-item-control"
        >
          <span
            class="ant-form-item-children"
          >
            <span
              class="ant-input-affix-wrapper"
            >
              <span
                class="ant-input-prefix"
              >
                <i
                  aria-label="icon: unordered-list"
                  class="anticon anticon-unordered-list"
                  style="color: rgba(0, 0, 0, 0.25);"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="unordered-list"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M912 192H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM104 228a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm0 284a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm0 284a56 56 0 1 0 112 0 56 56 0 1 0-112 0z"
                    />
                  </svg>
                </i>
              </span>
              <input
                class="ant-input"
                data-testid="form-input"
                placeholder="New task..."
                type="text"
                value=""
              />
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control-wrapper"
      >
        <div
          class="ant-form-item-control"
        >
          <span
            class="ant-form-item-children"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="submit"
            >
              <span>
                Create
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
